packages/webapp/components/invite/AISearchInvite.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { ReactElement, useEffect } from 'react';
+import { Button } from '@dailydotdev/shared/src/components/buttons/Button';
+import { ProfileImageLink } from '@dailydotdev/shared/src/components/profile/ProfileImageLink';
+import KeyIcon from '@dailydotdev/shared/src/components/icons/Key';
+import { useAuthContext } from '@dailydotdev/shared/src/contexts/AuthContext';
+import { AuthTriggers } from '@dailydotdev/shared/src/lib/auth';
+import { useMutation } from 'react-query';
+import { acceptFeatureInvitation } from '@dailydotdev/shared/src/graphql/features';
+import { useRouter } from 'next/router';
+import Logo, { LogoPosition } from '@dailydotdev/shared/src/components/Logo';
+import { useFeature } from '@dailydotdev/shared/src/components/GrowthBookProvider';
+import { feature } from '@dailydotdev/shared/src/lib/featureManagement';
+import { SearchExperiment } from '@dailydotdev/shared/src/lib/featureValues';
+import { useActions } from '@dailydotdev/shared/src/hooks/useActions';
+import { ActionType } from '@dailydotdev/shared/src/graphql/actions';
+import { cloudinary } from '@dailydotdev/shared/src/lib/image';
+import { campaignConfig, JoinPageProps } from './common';
+
+export function AISearchInvite({
+  referringUser,
+  token,
+}: JoinPageProps): ReactElement {
+  const router = useRouter();
+  const search = useFeature(feature.search);
+  const { completeAction } = useActions();
+  const { user, showLogin } = useAuthContext();
+  const { mutateAsync: onAcceptMutation } = useMutation(
+    acceptFeatureInvitation,
+    {
+      onSuccess: () => {
+        completeAction(ActionType.AcceptedSearch);
+        // TODO: check whether the feature will be set and forcibly apply it if not
+        router.push(campaignConfig.search.redirectTo);
+      },
+      onError: () => {
+        // TODO: display a toast based from the error response
+      },
+    },
+  );
+
+  const handleAcceptClick = () => {
+    if (!user) {
+      return showLogin(AuthTriggers.Author);
+    }
+
+    return onAcceptMutation({ token, referrerId: referringUser.id });
+  };
+
+  useEffect(() => {
+    if (search === SearchExperiment.Control) {
+      return;
+    }
+
+    router.push(campaignConfig.search.redirectTo);
+  }, [router, search]);
```

Comment:
I'm not sure this is correct.
The user got a invite for search correct, so surely they will be on control then?